openapi.yaml

```diff
@@ -471,6 +591,173 @@ paths:
           $ref: "#/components/responses/InternalServerError"
 components:
   schemas:
+    AddRelationArgs:
+      type: object
+      properties:
+        endpoints:
+          type: array
+          items:
+            type: string
+        via-cidrs:
+          type: array
+          items:
+            type: string
+      additionalProperties: false
+      required:
+        - endpoints
+    Application:
+      type: object
+      description: Information about a given application.
+      properties:
+        base:
+          $ref: '#/components/schemas/Base'
+        channel:
+          type: string
+        charm:
+          type: string
+        constraints:
+          $ref: '#/components/schemas/Constraints'
+        endpoint-bindings:
+          type: object
+          patternProperties:
+            .*:
+              type: string
+        exposed:
+          type: boolean
+        exposed-endpoints:
+          type: object
+          patternProperties:
+            .*:
+              $ref: '#/components/schemas/ExposedEndpoint'
+        life:
+          type: string
+        principal:
+          type: boolean
+        remote:
+          type: boolean
+        tag:
+          type: string
+      additionalProperties: false
+      required:
+        - tag
+        - principal
+        - exposed
+        - remote
+        - life
+    ApplicationSetCharm:
+      type: object
+      properties:
+        application:
+          type: string
+        channel:
+          type: string
+        charm-origin:
+          $ref: '#/components/schemas/CharmOrigin'
+        charm-url:
+          type: string
+        config-settings:
+          type: object
+          patternProperties:
+            .*:
+              type: string
+        config-settings-yaml:
+          type: string
+        endpoint-bindings:
+          type: object
+          patternProperties:
+            .*:
+              type: string
+        force:
+          type: boolean
+        force-base:
+          type: boolean
+        force-units:
+          type: boolean
+        generation:
+          type: string
+        resource-ids:
+          type: object
+          patternProperties:
+            .*:
+              type: string
```

Comment:
This feels like it needs more modelling - should this be a list of IDs? a list of resource objects?

Reply:
That's supposed to be a map of resource names to resource IDs, right? i.e., `map[string]string`, so I think the following should model it just fine, wdyt?:

```yaml
type: object
additonalProperties:
  type: string
```

Reply:
Perhaps it could be a list of `Resource` objects, where `Resource` has fields `name`, `id` and `type` (type is an `enum` containing `file` and `oci-image`)?